addons/im_livechat/models/im_livechat_channel.py

```diff
@@ -62,10 +71,48 @@ def _are_you_inside(self):
         for channel in self:
             channel.are_you_inside = self.env.user in channel.user_ids
 
-    @api.depends('user_ids.im_status')
+    @api.depends(
+        "user_ids.im_status",
+        "channel_ids.livechat_active",
+        "channel_ids.last_interest_dt",
+        "channel_ids.rtc_session_ids",
+    )
     def _compute_available_operator_ids(self):
+        session_count_by_operator = {}
+        operator_in_call = self.env["res.partner"]
+        if self.concurent_session:
+            old_livechat_dt = fields.Datetime.now() - timedelta(minutes=30)
+            session_count_by_operator = dict(
+                self.env["discuss.channel"]._read_group(
+                    expression.AND(
+                        [
+                            [("channel_type", "=", "livechat")],
+                            expression.OR(
+                                [
+                                    [("livechat_active", "=", True)],
+                                    [("last_interest_dt", "<=", old_livechat_dt)],
+                                ]
+                            ),
+                        ]
+                    ),
+                    groupby=["livechat_operator_id"],
+                    aggregates=["__count"],
+                )
+            )
```

Comment:
We're getting the count by operator for *every* operator. We should restrict this to the operators in `self.user_ids`. Additionally, we could further restrict the search by excluding operators that do not satisfy `_is_user_available`, thus avoiding unnecessary searches. Same remark for the query below.